modules/azure-landing-zone/frontdoor.tf

```diff
@@ -211,3 +191,39 @@ resource "azurerm_frontdoor" "main" {
 
   depends_on = [azurerm_frontdoor_firewall_policy.custom, azurerm_key_vault_access_policy.frontdoor_kv_access]
 }
+
+resource "azurerm_frontdoor_custom_https_configuration" "https" {
+  for_each = { for frontend in var.frontends :
+    frontend.name => frontend
+    if lookup(frontend, "enable_ssl", true)
```

Comment:
Just iterate through all the frontends and if enable_ssl is false then disable the https otherwise the default always is true i.e. enable https.